workflow/scripts/osemosys_global/powerplant/main.py

```diff
@@ -0,0 +1,299 @@
+import pandas as pd
+import os
+
+from read import(
+    import_plexos_2015,
+    import_weo_regions,
+    import_weo_costs,
+    import_op_life,
+    import_naming_convention_tech,
+    import_afs,
+    import_custom_res_cap,
+)
+
+from constants import(
+    DUPLICATE_TECHS,
+    MODE_LIST,
+    RENEWABLES_LIST,
+    DF_IAR_CUSTOM_VAL,
+    DF_OAR_CUSTOM_VAL
+    )
+
+from data import(
+    set_generator_table,
+    average_efficiency,
+    )
+
+from residual_capacity import(
+    res_capacity,
+    add_custom_res_cap
+    )
+
+from activity import(
+    activity_master_start,
+    activity_output_pwr,
+    activity_input_pwr,
+    activity_upstream,
+    activity_master_end,
+    capact
+    )
+
+from costs import(
+    costs_pwr,
+    costs_end
+    )
+
+from operational_life import set_op_life
+
+from investment_constraints import cap_investment_constraints
+
+from sets import(
+    create_sets,
+    output_sets
+    )
+
+from user_defined_capacity import set_user_defined_capacity
+
+from availability import availability_factor
+
+def main(
+    plexos_prop: pd.DataFrame,
+    plexos_memb: pd.DataFrame,
+    weo_costs: pd.DataFrame,
+    weo_regions: pd.DataFrame,
+    default_op_life: pd.DataFrame,
+    naming_convention_tech: pd.DataFrame,
+    custom_res_cap: pd.DataFrame,
+    default_av_factors: pd.DataFrame,
+):
+    
+    # CALL FUNCTIONS
+    
+    # return generator_table
+    gen_table = set_generator_table(plexos_prop, plexos_memb, op_life_dict, 
+                                    tech_code_dict, start_year, end_year)
+
+    # Calculate average technology efficiencies.
+    df_eff_node, df_eff_tech = average_efficiency(gen_table)
+
+    # Create master table for activity ratios.
+    df_ratios = activity_master_start(gen_table, DUPLICATE_TECHS, MODE_LIST,
+                                      custom_nodes, start_year, end_year)
```

Comment:
We can move this to a different issue, but in general we should probably avoid using global variables (ie. the `custom_nodes`, `start_year`, `end_year`). Or if we do want them to be global, move them to the constants file (but this is not what we want in this case, since these are configuration options)

Reply:
This has NOT been addressed with the fuel name fix !

Reply:
I agree but not sure how to get around it. Initially I had them in the constants file but since they are configurable that is not an option either (when run through snakemake).